fix compilation issues with Intel compilers
error: no suitable conversion function from "const boost::shared_ptr<BiasBoss>" to "bool" exists
author: Kenneth Hoste (HPC-UGent)
--- express-1.5.1-src/src/targets.cpp.orig	2018-01-23 17:01:41.507912423 +0100
+++ express-1.5.1-src/src/targets.cpp	2018-01-23 17:02:51.498864908 +0100
@@ -113,12 +113,12 @@
 
   double ll = LOG_1;
   double tot_mass = mass(with_pseudo);
-  double tot_eff_len = cached_effective_length(lib.bias_table);
+  double tot_eff_len = cached_effective_length(static_cast<bool>(lib.bias_table));
   if (neighbors) {
     foreach (const Target* neighbor, *neighbors) {
       tot_mass = log_add(tot_mass, neighbor->mass(with_pseudo));
       tot_eff_len = log_add(tot_eff_len,
-                            neighbor->cached_effective_length(lib.bias_table));
+                            neighbor->cached_effective_length(static_cast<bool>(lib.bias_table)));
     }
   }
   ll += tot_mass - tot_eff_len;
